services/provider/server/server.go

```diff
@@ -523,20 +538,84 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 		Name: "noobaa-remote-join-secret",
 		Kind: "Secret",
 		Data: mustMarshal(map[string]string{
-			"auth_token": string(authToken),
-			"mgmt_addr":  noobaaMgmtAddress,
+			"auth_token":            string(authToken),
+			"mgmt_addr":             fmt.Sprintf("https://%s:443", noobaaMgmtAddress),
+			"AWS_ACCESS_KEY_ID":     string(accessKeyID),
+			"AWS_SECRET_ACCESS_KEY": string(secretAccessKey),
 		}),
 	})
 
 	extR = append(extR, &pb.ExternalResource{
-		Name: "noobaa-remote",
+		Name: "noobaa",
 		Kind: "Noobaa",
 		Data: mustMarshal(&nbv1.NooBaaSpec{
 			JoinSecret: &v1.SecretReference{
 				Name: "noobaa-remote-join-secret",
 			},
 		}),
 	})
+
+	nb := &nbv1.NooBaa{}
+	nb.Name = "noobaa"
+	nb.Namespace = s.namespace
+
+	if err = s.client.Get(ctx, client.ObjectKeyFromObject(nb), nb); err != nil {
+		return nil, fmt.Errorf("failed to get noobaa %v", err)
+	}
+	if nb.Status.Phase != nbv1.SystemPhaseReady {
+		// Noobaa system is not yet ready
+		return extR, nil
+	}
+	nbS3Endpoint := nb.Status.Services.ServiceS3.NodePorts[0]
```

Comment:
@ezio-auditore Provided this is for internal usage and not to be exposed, can we also add the http endpoint (no SSL) if there is no security concern?

Reply:
Added  http port 80